spec/services/draw_activator_spec.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+RSpec.describe DrawActivator do
+  # TODO: extract this into a shared example; if we're going to use this pattern
+  # across all of our service objects we should have a way to easily implement
+  # it for all of them. Should be pretty easy - all we need to pass to the
+  # shared example is the method we want to call and the params hash
+  describe '.start' do
+    it 'calls :start on an instance of DrawStarter' do
+      draw = instance_spy('Draw')
+      draw_activator = create_mock_draw_activator(draw: draw)
+      described_class.activate(draw: draw)
+
+      expect(draw_activator).to have_received(:activate)
+    end
+  end
+
+  describe '#activate' do
+    it 'checks to make sure that the draw is a draft' do
+      draw = instance_spy('draw', draft?: false)
+
+      result = described_class.activate(draw: draw)
+
+      expect(result[:msg][:error]).to include('Draw must be a draft.')
+    end
+
+    it 'checks to make sure that the draw has at least one student' do
+      draw = instance_spy('draw', has_students?: false)
+
+      result = described_class.activate(draw: draw)
+
+      expect(result[:msg][:error]).to \
+        include('Draw must have at least one student.')
+    end
+
+    it 'checks to make sure that there are enough beds for students' do
+      draw = instance_spy('draw', has_enough_beds?: false)
+
+      result = described_class.activate(draw: draw)
+
+      expect(result[:msg][:error]).to \
+        include('Draw must have at least one bed per student')
+    end
+
+    it 'updates the status of the draw to pre_lottery' do
+      # draw = create_mock_draw
+      draw = create_draw_data
+
+      result = described_class.activate(draw: draw)
+
+      expect(result[:obj]).to be_pre_lottery
+    end
```

Comment:
Yea..... I don't think it solves my problem (the enum could change and this spec would still pass), but that may be fine as long as we have at least one integration test that _would_ fail if the enum changed, which is what we want. I was implicitly testing this by stubbing `update_attributes(status: 'pre_lottert')` to return the updated spy with `pre_lottery?: true` stubbed, but your assertion is better as we're really testing the message sent out of the service object, not the outcome.

That said, should we? I feel like we should only send the invitation e-mails if that update is successful, so as a sanity check should we make that action dependent on the result of `update_attributes`?